packages/pack/test/test.test.ts

```diff
@@ -0,0 +1,167 @@
+import { resolvePath } from "@typespec/compiler";
+import { createTester, expectDiagnosticEmpty } from "@typespec/compiler/testing";
+import { ok } from "assert";
+import { describe, expect, it } from "vitest";
+import { combineProjectIntoFile } from "../src/pack.js";
+import { d } from "./utils.js";
+
+const Tester = createTester(resolvePath(import.meta.dirname, ".."), { libraries: [] });
+describe("bundler", () => {
+  async function combine(files: Record<string, string>) {
+    const tester = await Tester.createInstance();
+    for (const [name, content] of Object.entries(files)) {
+      tester.fs.add(name, content);
+    }
+    return combineProjectIntoFile(tester.fs.compilerHost, "main.tsp");
+  }
+
+  async function combineTo(files: Record<string, string>): Promise<string> {
+    const result = await combine(files);
+    expectDiagnosticEmpty(result.diagnostics);
+    ok(result.content);
+    return result.content;
+  }
+
+  it("insert imported local files", async () => {
```

Comment:
Should add tests:
- emits diagnostic for js file imports
- combines namespaces across files (same named namespace used in multiple files)
- (maybe) handles file with multiple namespaces (`namespace A {   }`  `namespace B {   }`)